core/server/models/plugins/filter.js

```diff
@@ -1,92 +1,165 @@
+     * defaults - filters which must be applied if a matching filter isn't provided
+     * custom - custom filters which are additional
+     * extra - filters coming from model filter aliases
+     */
+    mergeFilters: ({enforced, defaults, custom, extra} = {}) => {
+        if (extra) {
+            // NOTE: check if custom and extra are treated as 'and'?
+            if (custom) {
+                custom += `+${extra}`;
+            } else {
+                custom = extra;
+            }
         }
 
-        // Reduce custom filters based on enforced filters
-        if (custom && !_.isEmpty(custom.statements) && enforced && !_.isEmpty(enforced.statements)) {
-            custom.statements = gql.json.rejectStatements(custom.statements, function (customStatement) {
-                return gql.json.findStatement(enforced.statements, customStatement, 'prop');
-            });
+        if (custom && !enforced && !defaults) {
+            return custom;
         }
 
-        // Reduce default filters based on custom filters
-        if (defaults && !_.isEmpty(defaults.statements) && custom && !_.isEmpty(custom.statements)) {
-            defaults.statements = gql.json.rejectStatements(defaults.statements, function (defaultStatement) {
-                return gql.json.findStatement(custom.statements, defaultStatement, 'prop');
-            });
+        let merged = '';
+
+        if (enforced) {
+            merged += enforced;
         }
 
-        // Merge enforced and defaults
-        enforced = gql.json.mergeStatements(enforced, defaults);
+        if (custom) {
+            custom = filterUtils.reduceFilters(merged, custom);
 
-        if (_.isEmpty(custom.statements)) {
-            return enforced;
+            if (custom) {
+                merged = merged ? `${merged}+${custom}` : custom;
+            }
         }
 
-        if (_.isEmpty(enforced.statements)) {
-            return custom;
+        if (defaults) {
+            defaults = filterUtils.reduceFilters(merged, defaults);
+
+            if (defaults) {
+                merged = merged ? `${merged}+${defaults}` : defaults;
+            }
         }
 
-        return {
-            statements: [
-                {group: enforced.statements},
-                {group: custom.statements, func: 'and'}
-            ]
-        };
+        return merged;
+    },
+
+    /**
+     * ## Process Filters
+     * Util that substitutes aliases and expands expressions with custom filters
+     *
+     */
+    processFilters: (filters, aliases) => {
+        let processed = filters;
+
+        aliases.forEach((alias) => {
+            const {key, replacement, filter} = alias;
+            if (processed.match(key)) {
+                // matches:
+                // - 'key:customFilter'
+                // - 'key:[customFilter]'
+                // - 'key:[custom,filter]'
+                const keyGroupMatch = `${key}:\\s?(\\w+|\\[(\\w+|,)+\\])`;
```

Comment:
The regex will not work for e.g. `primary_author:joe-blog`

We need e.g.
```
`${key}:\\s?([a-zA-Z\\-]+|\\[([a-zA-Z\\-]+|,)+\\])
```

Reply:
good. point. also looking at it with a fresh eye, need to allow for negation operator as well